cmd/client_test.go

```diff
@@ -95,59 +101,67 @@ func generateRedisProbeDef(name string) *probes_configpb.ProbeDef {
 
 // TestRegisterAndAddProbe tests that an extension probe type can be registered and added to Cloudprober without error.
 func TestRegisterAndAddProbe(t *testing.T) {
-	testHermes, client := setupCloudproberAndClient()
-	defer teardownCloudproberAndClient(testHermes, client)
+	testHermes, client := setupCloudproberAndClient(t)
+	defer teardownCloudproberAndClient(t, testHermes, client)
 
-	for i := 0; i < len(testProbes); i++ {
-		err := client.RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), testHermes.Ctx, generateRedisProbeDef(testProbes[i]), &myprobe.Probe{})
-		if err != nil {
-			t.Errorf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
+	for i := 0; i < testRunCount; i++ {
+		probe := fmt.Sprintf("%s%d", "testExtension", i)
+		if err := client.RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), testHermes.Ctx, generateRedisProbeDef(probe), &myprobe.Probe{}); err != nil {
+			t.Fatalf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
 		}
 
 		// Check if the probe was added correctly
-		listProbesResp, err := client.ListProbes()
+		listProbesResp, err := client.ListProbes(testHermes.Ctx)
 		if err != nil {
-			t.Errorf("ListProbes() failed, expected error: %v, got: %v", nil, err)
+			t.Fatalf("ListProbes() failed, expected error: %v, got: %v", nil, err)
 		}
 
 		probesList := listProbesResp.GetProbe()
-		if probesList[i].GetName() != testProbes[i] {
-			t.Errorf("Expected probe %s to be running, got probe %s running", testProbes[i], probesList[i].GetName())
+
+		// Sorts the probes by their extension number as they are in the format:
+		// "testExtension" + <number>, e.g. "testExtension0".
+		sort.Slice(probesList[:], func(i, j int) bool {
```

Comment:
Suspect you want `sort.SliceStable`

Reply:
Thank you! This is perfect, I've change this to `sort.SliceStable`.